nat-lab/tests/test_notification_center.py

```diff
@@ -35,23 +50,96 @@ def verify_mqtt_payload(payload):
         verify_uuid(machine_uuid)
 
 
-def run_mqtt_listener():
-    payload_queue: queue.Queue[str] = queue.Queue(1)
-
-    def on_message(_client, _userdata, message):
-        payload_queue.put(f"{message.payload.decode()}")
-
-    mqttc = mqtt.Client(client_id="reciever", protocol=mqtt.MQTTv311)
-
-    mqttc.on_message = on_message
-
-    mqttc.tls_set(certfile=None, keyfile=None, cert_reqs=ssl.CERT_NONE)
-    mqttc.connect(MQTT_BROKER_IP, port=8883, keepalive=1)
-    mqttc.subscribe("meshnet", qos=0)
-
-    mqttc.loop_start()
-
-    return payload_queue, mqttc
+async def run_mqtt_listener(
+    exit_stack,
+    connection,
+    mqtt_broker_host,
+    mqtt_broker_port,
+    mqtt_broker_user,
+    mqtt_broker_password,
+):
+    mqtt_process = await exit_stack.enter_async_context(
+        connection.create_process([
+            "python3",
+            "/opt/bin/mqtt-listener.py",
+            mqtt_broker_host,
+            mqtt_broker_port,
+            mqtt_broker_user,
+            mqtt_broker_password,
+        ]).run()
+    )
+    return await wait_for_mqtt_executing(mqtt_process)
+
+
+def wait_with_timeout(max_wait=5, initial_wait_time=0.1, factor=2, on_timeout=None):
```

Comment:
Yeah, it may look overcomplicated for such a simple test, but we will have a clear indication of what failed. If we have just basic sleep, then the pytest session may fail with the timeout in case process doesn't return anything.

Reply:
After discussion offline, we decided not to overcomplicate this test and use basic `while with sleep`